persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/PatientMapper.xml

```diff
@@ -19,4 +19,57 @@
             </include>
         </if>
     </sql>
+
+    <sql id="from">
+        FROM patient
+        INNER JOIN cancer_study ON patient.CANCER_STUDY_ID = cancer_study.CANCER_STUDY_ID
+    </sql>
+
+    <sql id="where">
+        <where>
+            <if test="patientIds == null">
+                cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[0]}
```

Comment:
Yes it should throw an error, there should be lots of other error checks, but currently I implement it only for the web layer inputs, since persistence module it's not going to be used by other 3rd party software in a foreseeable future. So my priority is to release these basic APIs in a timely manner, instead of making them impossible to break for 3rd party use. So I say we better leave it for later as well.